lib/reporters/TapReporter.js

```diff
@@ -116,7 +119,30 @@ function prettyYamlValue (value, indent = 4) {
   }
 
   // Handle null, boolean, array, and object
-  return JSON.stringify(value, null, 2);
+  return JSON.stringify(value, createCycleSafeReplacer(), 2);
+}
+
+/**
+ * Creates a replacer function to use with JSON.stringify that
+ * safely handles cyclical references. When a cycle is detected
+ * it will be represented as "[Circular]".
+ */
+function createCycleSafeReplacer() {
+  const seen = new Set();
+
+  return function cycleSafeReplacer(_key, value) {
+    let result = value;
+
+    if (typeof value === 'object' && value !== null) {
+      if (seen.has(value)) {
+        return '[Circular]';
```

Comment:
node’s util.inspect tracks specific refs here, so you know what it’s circular *of* - it might be worth doing that here.

Reply:
This is indeed a neat feature of `inspect`. One question is whether this would be confusing seeing it in the actual/expected output. To label the refs we need two parts:

1. To label the object being referenced as a `<ref *n>`
2. To use the correct index when labeling cycles `[Circular *n]`

Labeling the cycles should be pretty easy, but I'm a little worried about changing the structure of the stringified object too much. 

There are some other enhancements to this stringification we need to consider, for example, `BigInt`. I'd rather follow up on enhancements separately and landing this fix first.

Thoughts?

Reply:
Right - normal JSON.stringify isn't really an option, I think, at least not without a stringifier function also provided.